python-lib/spell_checker/dataframe_helpers.py

```diff
@@ -0,0 +1,50 @@
+def create_new_column_name(col_name, suffix, dataset_columns):
+    """
+    input:
+        - col_name: column name of the dataset df
+        - suffix: suffix to be appended to col_name
+        - dataset_columns: list of all columns of dataset
+    output:
+        - new column name with suffix that is not in the columns of the dataset
+    To ensure the output column name is not in the dataset, we append a number to col_name + "_" + suffix if needed
+    """
+    if col_name + "_" + suffix in dataset_columns:
+        i = 0
+        col_new_name = col_name + "_" + suffix + "_" + str(i)
+
+        while col_new_name in dataset_columns:
+            i += 1
+            col_new_name = col_name + "_" + suffix + "_" + str(i)
+        return col_new_name
+    else:
+        return col_name + "_" + suffix
+
+
+def create_all_new_column_names(df_columns, text_col_list):
```

Comment:
We'll need to add a column with the extracted misspellings (as a list), and a column with the count of mispellings